### PR TITLE
fixed missing extract_path

### LINKED ISSUE #1466: [BUG] Automatic downloader for sktime.datasets.load_UCR_UEA_dataset does not work for custom extract_path
Even given a custom extract_path, UCR_UEA datasets are downloaded to the default sktime package directory. The bug is easily fixed by adding the custom extract_path to the _download_and_extract function call.

Line 175 in sktime/sktime/datasets/_data_io.py in function load_UCR_UEA_dataset:

```python
try:
     _download_and_extract(
                url,
      )
```

should be instead

```python
try:
     _download_and_extract(
                url,
                extract_path=extract_path
      )
```

Expected behaviour:

Dataset gets downloaded to the correct path after bug fix.

### PR DESCRIPTION
**Pull request by yairbeer**
#### Reference Issues/PRs
Fixes #1466

#### What does this implement/fix? Explain your changes.
Added missing extract_path in line 178 in sktime/datasets/_data_io.py to fix the bug

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by yairbeer** [comment]
Hi, I've tested the branch locally and it passes. I'm not sure how should I add the test for the custom folder. Should I create a custom folder for extract inside test, load a custom dataset there, and check that it exists?

**Comment by fkiraly** [comment]
>Should I create a custom folder for extract inside test, load a custom dataset there, and check that it exists?

Yes, that would work, though the problem would be to do this that it works on all the operating systems.

Previously, we had a lot of problems with tests that were downloading files somewhere, setting off virus scanners or accidentally cluttering system folders.
So, in this case, I'd exceptionally say that it's perhaps better to avoid testing this, unless we can come up with a robust way to avoid all the unintended consequences...

**Review by fkiraly — APPROVED** [review]
Looks good!

While we wait whether someone else looks at it, would you mind changing the contributor names to GitHub IDs? That's better for the automated doc generation.

File: sktime/datasets/_data_io.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "Sebastiaan Koel",
     "Emilia Rose",
     "Tony Bagnall",
+    "Yair Beer",
```
Comment (fkiraly) [inline sktime/datasets/_data_io.py:38]: please use GitHub IDs (and change the other names to GitHub IDs too if you don't mind)

not a merge blocker

Comment (yairbeer) [reply sktime/datasets/_data_io.py]: Done, I used .all-contributorsrc for the IDs.

**Comment by yairbeer** [comment]
070     >>> from sktime.classification.feature_based import TSFreshClassifier
071     >>> from sktime.datasets import load_italy_power_demand
072     >>> X_train, y_train = load_italy_power_demand(split="train", return_X_y=True)
073     >>> X_test, y_test = load_italy_power_demand(split="test", return_X_y=True)
074     >>> clf = TSFreshClassifier()
075     >>> clf.fit(X_train, y_train)
UNEXPECTED EXCEPTION: TypeError("acf() got an unexpected keyword argument 'unbiased'")
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/site-packages/tsfresh/utilities/distribution.py", line 43, in _function_with_partly_reduce
    results = list(itertools.chain.from_iterable(results))
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/site-packages/tsfresh/utilities/distribution.py", line 42, in <genexpr>
    results = (map_function(chunk, **kwargs) for chunk in chunk_list)
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/site-packages/tsfresh/feature_extraction/extraction.py", line 337, in _do_extraction_on_chunk
    return list(_f())
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/site-packages/tsfresh/feature_extraction/extraction.py", line 317, in _f
    result = func(x, param=parameter_list)
  File "/opt/_internal/cpython-3.7.12/lib/python3.7/site-packages/tsfresh/feature_extraction/feature_calculators.py", line 404, in agg_autocorrelation
    a = acf(x, unbiased=True, fft=n > THRESHOLD_TO_USE_FFT, nlags=max_maxlag)[1:]
TypeError: acf() got an unexpected keyword argument 'unbiased'

This error failed the build (more than once), any idea why it happened?

Everything passed locally

**Comment by fkiraly** [comment]
Hm, looks like this is coming from an interface change in `statsmodels`, as a `tsfresh` dependency?

Where `tsfresh` is calling `stattools.acf` which no longer has an `unbiased` argument in 0.13.0, but used to have?

**Comment by fkiraly** [comment]
I've opened an issue for the bug  #1478, it seems unrelated to your contribution

**Review by fkiraly — APPROVED** [review]
looks good again, with the statsmodels issue fixed

**Pull request merged**